lib/Drush/Sql/Sqlmysql.php

```diff
@@ -32,7 +32,9 @@ public function creds($hide_password = TRUE) {
 
     // Some drush commands (e.g. site-install) want to connect to the
     // server, but not the database.  Connect to the built-in database.
-    $parameters['database'] = empty($this->db_spec['database']) ? 'information_schema' : $this->db_spec['database'];
+    if (!empty($this->db_spec['database'])) {
```

Comment:
I'm a little hesitant to remove information_schema. I think we need to specify a DB in certain cases (see early parts of site-install command). Needs investigation.